scripts/check_imports_and_dependencies.py

```diff
@@ -90,7 +90,20 @@ def get_imports_for_file(pyfile: Union[str, Path]) -> List[str]:
         with open(pyfile, "r") as f:
             statements = f.read()
         instructions = dis.get_instructions(statements)  # type: ignore
-        imports = [i for i in instructions if "IMPORT" in i.opname]
+        imports = []
+        is_try_except_block = False
+        for im in instructions:
+            if "SETUP_FINALLY" in im.opname:
+                is_try_except_block = True
+
+            if "POP_EXCEPT" in im.opname:
+                is_try_except_block = False
+
+            if is_try_except_block:
+                continue
+
+            if "IMPORT" in im.opname:
+                imports.append(im)
```

Comment:
So we skip import statements made during a `try/except` block

Reply:
please add a comment to the code so we know why next time